packages/middleware-sdk-s3/src/write-get-object-response-endpoint.ts

```diff
@@ -0,0 +1,83 @@
+import { getSuffixForArnEndpoint } from "@aws-sdk/middleware-bucket-endpoint";
+import { HttpRequest } from "@aws-sdk/protocol-http";
+import {
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOutput,
+  BuildMiddleware,
+  HandlerExecutionContext,
+  MetadataBearer,
+  Pluggable,
+  Provider,
+  RelativeMiddlewareOptions,
+} from "@aws-sdk/types";
+
+type PreviouslyResolved = {
+  region: Provider<string>;
+  isCustomEndpoint: boolean;
+  disableHostPrefix: boolean;
+  runtime: string;
+};
+
+type Input = {
+  RequestRoute: string | undefined;
+};
+
+/**
+ * @internal
+ */
+export const writeGetObjectResponseEndpointMiddleware =
+  (config: PreviouslyResolved): BuildMiddleware<any, any> =>
+  <Output extends MetadataBearer>(
+    next: BuildHandler<Input, Output>,
+    context: HandlerExecutionContext
+  ): BuildHandler<Input, Output> =>
+  async (args: BuildHandlerArguments<Input>): Promise<BuildHandlerOutput<Output>> => {
+    const { region: regionProvider, isCustomEndpoint, disableHostPrefix } = config;
+    const region = await regionProvider();
+    const { request, input } = args;
+    if (!HttpRequest.isInstance(request)) return next({ ...args });
+    let hostname = request.hostname;
+    if (hostname.endsWith("s3.amazonaws.com") || hostname.endsWith("s3-external-1.amazonaws.com")) {
```

Comment:
are these defined by anything in the config, as opposed to hard-coding it?

Reply:
They are defined in the endpoints.json: https://github.com/aws/aws-sdk-js-v3/blob/efdc6a062180b7ca5fc4ad149c50800f86bb1268/codegen/smithy-aws-typescript-codegen/src/main/resources/software/amazon/smithy/aws/typescript/codegen/endpoints.json#L11249. It's not available in config. Hardcoding it as it's safer than Regex. We don't expect S3 to launch more global regions anytime soon.